eclair-core/src/main/scala/fr/acinq/eclair/db/sqlite/SqlitePaymentsDb.scala

```diff
@@ -225,4 +241,26 @@ class SqlitePaymentsDb(sqlite: Connection) extends PaymentsDb with Logging {
     }
   }
 
+  private def insertFailures(id: UUID, failures: Traversable[String]): Unit = {
+    failures.filter(_.nonEmpty).foreach { failure =>
+      using(sqlite.prepareStatement("INSERT INTO sent_payments_failures (id, failure) VALUES (?, ?)")) { statement =>
+        statement.setString(1, id.toString)
+        statement.setString(2, failure)
+        val res = statement.executeUpdate()
+      }
+    }
+  }
+
+  private def selectFailures(id: UUID): Seq[String] = {
+    val res = new mutable.ListBuffer[String]
+    using(sqlite.prepareStatement("SELECT failure FROM sent_payments_failures WHERE id = ?")) { statement =>
+      statement.setString(1, id.toString)
+      val rs = statement.executeQuery()
+      while (rs.next()) {
+        res += rs.getString(1)
```

Comment:
nit: for consistency getters use string based parameters
```suggestion
        res += rs.getString("failure")
```